### PR TITLE
feat(SHOPIFY-158): add create method for products

### PR DESCRIPTION
**Pull request by connor-mohc**

### CODE REVIEW

File: images.go [hunk 1]
```diff
@@ -8,11 +8,11 @@ import (
...
-	CreatedAt time.Time `json:"created_at"`
-	SRC       string    `json:"src"`
-	Width     int       `json:"width"`
-	Height    int       `json:"height"`
-	Alt       string    `json:"alt"`
+	CreatedAt time.Time `json:"created_at,omitempty"`
+	SRC       string    `json:"src,omitempty"`
+	Width     int       `json:"width,omitempty"`
+	Height    int       `json:"height,omitempty"`
+	Alt       string    `json:"alt,omitempty"`
```
Comment (harry-mohc) [inline images.go:15]: I feel like mandatory fields shouldnt have this omit empty flag on right? Have you checked which fields are mandatory?

File: product-imges.go [hunk 2]
```diff
@@ -36,6 +36,16 @@ func (dtos ProductImageDTOs) ToShopify() shopify.ProductImages {
 	return productImages
 }
 
+func buildProductImageDTOs(productImages shopify.ProductImages) ProductImageDTOs {
```
Comment (harry-mohc) [inline product-imges.go:39]: this should be an exported function like the other dto builders

**Review by Thomas-Richards-MoHC — APPROVED** [review]

File: product-imges.go [hunk 3]
```diff
@@ -96,6 +106,17 @@ func (dto ProductImageDTO) ToShopify() shopify.ProductImage {
 	}
 }
 
+func buildProductImageDTO(productImage shopify.ProductImage) ProductImageDTO {
```
Comment (harry-mohc) [inline product-imges.go:109]: this should also be exported

File: variants.go [hunk 4]
```diff
@@ -36,16 +36,26 @@ func (dtos VariantDTOs) ToShopify() shopify.Variants {
 	return variants
 }
 
+func buildVariantDTOs(variants shopify.Variants) VariantDTOs {
```
Comment (harry-mohc) [inline variants.go:39]: and this

**Review by harry-mohc — CHANGES_REQUESTED** [review]

File: product-imges.go (hunk 2 above)
Comment (connor-mohc) [reply product-imges.go:39]: Ah the `dto` builder I looked at in `line-items` which is probably the only one not exported :sweat_smile:

Comment (harry-mohc) [reply product-imges.go:39]: oooh would you mind fixing that one?

File: product-imges.go [hunk 5]
```diff
@@ -36,6 +36,16 @@ func (dtos ProductImageDTOs) ToShopify() shopify.ProductImages {
 	return productImages
 }
 
+func BuildProductImageDTOs(productImages shopify.ProductImages) ProductImageDTOs {
```
Comment (Thomas-Richards-MoHC) [inline product-imges.go:39]: now these functions need comments

File: products.go [hunk 6]
```diff
@@ -22,6 +22,53 @@ func newProductRepository(client http.Client, createURL func(endpoint string) st
 	}
 }
 
+func (repository productRepository) Create(product shopify.Product) (shopify.Product, error) {
+	createDTO := ProductDTO{
```
Comment (harry-mohc) [inline products.go:26]: why not use the build dto function?

**Review by harry-mohc — CHANGES_REQUESTED** [review]

**Review by harry-mohc — APPROVED** [review]

**Pull request merged**